splunk/packages/edit-table/src/components/KVStoreUploader.jsx

```diff
@@ -21,77 +27,126 @@ const ModalButtonActionGroup = styled.div`
+            message,
+        });
+    };
+
     const onCloseUploadModal = () => {
         setUploadModalOpen(false);
         setUploadedCSVContent(undefined);
     };
     const handleFileChange = (fileContent) => setUploadedCSVContent(fileContent);
-    const handleUploadCSV = async () => {
-        if (tableMetadata == null) {
-            return;
+
+    const transformData = async () => {
+        if (uploadedCSVContent == null) {
+            showErrorMessage(emptyErrorMsg);
+            return false;
         }
 
-        setUploading(true);
-        const { dataFields, totalItems } = tableMetadata;
-        const emptyErrorMsg = 'No data to upload.';
-        const emptyCSVErrorMsg = 'CSV file is empty.';
-        const dataFieldErrorMsg = 'CSV file have no enough data fields.';
-        const deleteErrorMsg = 'Error deleting all entries. Please try again.';
-        const uploadErrorMsg = 'Error uploading csv. Please try again.';
+        const jsonData = await convertToJSONArrayFromCSVString(uploadedCSVContent);
+        if (jsonData == null || jsonData.length === 0) {
+            showErrorMessage(emptyCSVErrorMsg);
+            return false;
+        }
 
-        try {
-            if (uploadedCSVContent == null) {
-                throw new Error(emptyErrorMsg);
-            }
+        const isCorrectFormat = checkJsonArrayCorrectFormat(jsonData, dataFields);
+        if (!isCorrectFormat) {
+            showErrorMessage(dataFieldErrorMsg);
+            return false;
+        }
 
-            const jsonData = await convertToJSONArrayFromCSVString(uploadedCSVContent);
-            if (jsonData == null || jsonData.length === 0) {
-                throw new Error(emptyCSVErrorMsg);
-            }
+        const formattedJsonData = projectFields(jsonData, dataFields);
+        return formattedJsonData;
+    };
 
-            const isCorrectFormat = checkJsonArrayCorrectFormat(jsonData, dataFields);
-            if (!isCorrectFormat) {
-                throw new Error(dataFieldErrorMsg);
+    const batchInsertCollectionEntries = (newData) => {
+        const promises = [];
+        for (let i = 0; i < newData.length; i += BATCH_SIZE) {
+            const chunk = newData.slice(i, i + BATCH_SIZE);
+            promises.push(
+                insertCollectionEntries(splunkApp, collectionName, chunk, uploadErrorMsg)
+            );
+        }
+        return promises;
+    };
+
+    const doKvStoreChanges = async (callback) => {
+        await createBackupForKvStore(splunkApp, kvStore, backupErrorMsg);
+        try {
+            await callback();
+        } catch (changeError) {
+            try {
+                await restoreKvStoreFromBackup(splunkApp, kvStore, restoreErrorMsg);
+            } catch (restoreError) {
+                throw new Error(
+                    `Update Error: ${changeError.message}. Backup Error: ${restoreError.message}`
+                );
             }
+            throw changeError;
+        }
+    };
```

Comment:
how about moving this to the `data` file? The code is component independent and you won't have to import create and restore backup functions then here